confero-api/src/main/java/org/zpi/conferoapi/application/ApplicationService.java

```diff
@@ -0,0 +1,106 @@
+
+@Service
+@RequiredArgsConstructor
+@FieldDefaults(makeFinal = true, level = lombok.AccessLevel.PRIVATE)
+public class ApplicationService {
+
+    ConferenceEditionRepository conferenceEditionRepository;
+    UserRepository userRepository;
+    PresenterRepository presenterRepository;
+    SessionRepository sessionRepository;
+    PresentationRepository presentationRepository;
+    OrcidService orcidService;
+    SecurityUtils securityUtils;
+
+
+    public Session createApplication(CreateApplicationRequest request) {
+        var currentUser = securityUtils.getCurrentUser();
+        var activeConferenceEdition = conferenceEditionRepository.findActiveEditionConference();
+
+        if (activeConferenceEdition.isEmpty()) {
+            throw new ServiceException(NO_ACTIVE_CONFERENCE_EDITION);
+        }
+
+        var session = Session.builder()
+                .title(request.getTitle())
+                .type(request.getType())
+                .creator(currentUser)
+                .tags(request.getTags())
+                .edition(activeConferenceEdition.get())
+                .description(request.getDescription())
+                .status(request.getSaveAsDraft() ? ApplicationStatus.DRAFT : ApplicationStatus.PENDING)
+                .createdAt(Instant.now())
+                .presentations(new ArrayList<>())
+                .build();
+
+        sessionRepository.save(session);
+
+        request.getPresentations().forEach(presentationRequest -> {
+
+            var newPresentation = Presentation.builder()
+                    .title(presentationRequest.getTitle())
+                    .session(session)
+                    .presenters(new ArrayList<>())
+                    .build();
+
+            var savedPresentation = presentationRepository.save(newPresentation);
+
+
+            presentationRequest.getPresenters().forEach(presenter -> {
+                var existingUser = userRepository.findByEmailOrOrcid(presenter.getEmail(), presenter.getOrcid())
+                        .orElseGet(() ->
+                                userRepository.save(
+                                        User.builder()
+                                                .isAdmin(false)
+                                                .email(presenter.getEmail())
+                                                .build()
+                                )
+                        );
+
+                var presenterInfo = orcidService.getRecord(presenter.getOrcid());
+
+                Presenter newPresenter = Presenter.builder()
+                        .email(presenter.getEmail())
+                        .presentation(savedPresentation)
+                        .orcid(presenter.getOrcid())
+                        .name(presenterInfo.getName())
+                        .surname(presenterInfo.getSurname())
+                        .title(presenterInfo.getTitle())
+                        .organization(presenterInfo.getOrganization())
```

Comment:
I expect concurrent calls to the ORCID api here. CompletableFuture can be used

Reply:
Or, instead, this can be handled in the controller(concurrent requests), with all presenter data being passed to the service.